src/book.ts

```diff
@@ -0,0 +1,46 @@
+import { Item } from './item';
+import { Pages } from './pages';
+
+export class Book extends Item {
+    title: string;
+    author: string;
+    pages: Pages;
+
+    constructor(title: string, author: string, pages: Pages) {
+        super();
+        this.title = title;
+        this.author = author;
+        this.pages = pages;
+    }
+
+    setTitle(title: string) {
+        this.title = title;
+    }
+
+    getTitle(): string {
+        return this.title;
+    }
+
+    setAuthor(author: string) {
+        this.author = author;
+    }
+
+    getAuthor(): string {
+        return this.author;
+    }
+
+    toString(): string {
+        return `Book: ${this.title} by ${this.author} with number of pages: ${this.pages}`;
```

Comment:
this.pages is an instance of class Pages, you won't get the number of pages like this.
You should either call this.pages.array.length or (a better option) create a method in Pages class to get the number of pages (array's length).